pallets/interest-accrual/src/lib.rs

```diff
@@ -240,7 +259,9 @@ pub mod pallet {
 			interest_rate_per_sec: T::InterestRate,
 			normalized_debt: T::Balance,
 		) -> Result<T::Balance, DispatchError> {
-			Rate::<T>::get(interest_rate_per_sec)
+			Rates::<T>::get()
+				.into_iter()
+				.find(|rate| rate.interest_rate_per_sec == interest_rate_per_sec)
```

Comment:
Seems like this can be refactored into a `get_rate(interest_rate_per_sec: Rate) -> Option<RateDetailsOf<T>>` method, as the code is re-used multiple times.

Reply:
Yeah. I think in my mind this was a "one-liner", but its enough stuff, and common enough, to be worth giving a name :+1: